std/datetime/stopwatch.d

```diff
@@ -0,0 +1,425 @@
+    }
+
+    @safe nothrow @nogc unittest
+    {
+        assert(StopWatch.init.peek() == Duration.zero);
+    }
+
+
+    /++
+       Sets the total time which the StopWatch has been running (i.e. what peek
+       returns).
+
+       The StopWatch does not have to be stopped for setTimeElapsed to be
+       called, nor will calling it cause the StopWatch to stop.
+      +/
+    void setTimeElapsed(Duration timeElapsed) @safe nothrow @nogc
+    {
+        enum hnsecsPerSecond = convert!("seconds", "hnsecs")(1);
+        _ticksElapsed = convClockFreq(timeElapsed.total!"hnsecs", hnsecsPerSecond, MonoTime.ticksPerSecond);
+        _timeStarted = MonoTime.currTime;
+    }
+
+    ///
+    @system nothrow @nogc unittest
+    {
+        import core.thread : Thread;
+
+        StopWatch sw;
+        sw.setTimeElapsed(hours(1));
+
+        // As discussed in MonoTime's documentation, converting between
+        // Duration and ticks is not exact, though it will be close.
+        // How exact it is depends on the frequency/resolution of the
+        // system's monotonic clock.
+        assert(abs(sw.peek() - hours(1)) < usecs(1));
+
+        sw.start();
+        Thread.sleep(usecs(1));
+        assert(sw.peek() > hours(1) + usecs(1));
+    }
+
+
+    /++
+       Returns whether this StopWatch is currently running.
+      +/
+    @property bool running() @safe const pure nothrow @nogc
+    {
+        return _running;
+    }
+
+    ///
+    @safe nothrow @nogc unittest
+    {
+        StopWatch sw;
+        assert(!sw.running);
+        sw.start();
+        assert(sw.running);
+        sw.stop();
+        assert(!sw.running);
+    }
+
+
+private:
+
+    // We track the ticks for the elapsed time rather than a Duration so that we
+    // don't lose any precision.
+
+    bool _running = false; // Whether the StopWatch is currently running
+    MonoTime _timeStarted; // The time the StopWatch started measuring (i.e. when it was started or reset).
+    long _ticksElapsed;    // Total time that the StopWatch ran before it was stopped last.
+}
+
+
+/++
+    Benchmarks code for speed assessment and comparison.
+
+    Params:
+        fun = aliases of callable objects (e.g. function names). Each callable
+              object should take no arguments.
+        n   = The number of times each function is to be executed.
+
+    Returns:
+        The amount of time (as a $(REF Duration,core,time)) that it took to call
+        each function $(D n) times. The first value is the length of time that
+        it took to call $(D fun[0]) $(D n) times. The second value is the length
+        of time it took to call $(D fun[1]) $(D n) times. Etc.
+  +/
+Duration[fun.length] benchmark(fun...)(uint n)
+{
+    Duration[fun.length] result;
+    auto sw = StopWatch(AutoStart.yes);
+
+    foreach (i, unused; fun)
+    {
+        sw.reset();
+        foreach (_; 0 .. n)
+            fun[i]();
+        result[i] = sw.peek();
+    }
+
+    return result;
```

Comment:
I probably removed it with the idea that it did the same thing as the defaults, since when `StopWatch` was originally added, it was not uncommon that you had to declare it yourself. Looking over it now, it looks like the original implementation is different from the default in that it skips comparing the `_running` member, therefore not treating whether the `StopWatch` is running as part of its comparable state. I don't know if that makes sense or not. I strongly question whether it makes any sense to compare a `StopWatch` in the first place.